R/profile.R

```diff
@@ -0,0 +1,455 @@
+    signifAbove <- ret[[which]] > currentPar
+    signifBelow <- ret[[which]] < currentPar
+    convergedSignif <- any(signifSame & signifAbove) && any(signifSame & signifBelow)
+
+    currentOFVDiff <- min(abs(targetOfv - ret$OFV), na.rm = TRUE)
+    convergedOfv <- currentOFVDiff <= ofvtol
+
+    converged <- convergedSignif | convergedOfv
+
+    if (converged) {
+      if (convergedSignif & convergedOfv) {
+        messageProfileComplete(which, direction = direction, "complete due to significant digits precision and achieving OFV within specified tolerance")
+      } else if (convergedSignif) {
+        messageProfileComplete(which, direction = direction, "complete due to significant digits precision")
+      } else if (convergedOfv) {
+        messageProfileComplete(which, direction = direction, "complete due to achieving OFV within specified tolerance")
+      } else {
+        stop("Unclear convergece criteria, please report a bug") # nocov
+      }
+      storeBound <- data.frame(Parameter = which, X = currentPar, profileBound = direction*ofvIncrease)
+      names(storeBound)[2] <- which
+      ret <- dplyr::bind_rows(storeBound, ret)
+      currentIter <- Inf
+    } else {
+      message("Profiling ", which, " best distance to target OFV: ", currentOFVDiff)
+    }
+  }
+  if (is.finite(currentIter)) {
+    messageProfileComplete(which, direction = direction, "aborted due to too many iterations without convergence")
+  }
+
+  # Sort the output by parameter value
+  ret <- ret[order(ret[[which]]), ]
+  # Add extra information to assist with subsequent
+  # The output rownames are uninformative and distracting; remove them.
+  rownames(ret) <- NULL
+  ret
+}
+
+#' Control options for log-likelihood profiling
+#'
+#' @param ofvIncrease The targetted change in objective function value (3.84
+#'   corresponds to a Chi-squared test with a 95% confidence interval)
+#' @param rseTheta The relative standard error (percent) for the model
+#'   parameters.  It can be missing (the default) in which case a default value
+#'   of 30% will be applied.  If given as a single number, it will be applied to
+#'   all parameters.  If given as a named vector of numbers, it will be applied
+#'   to each named parameter.
+#' @param itermax Maximum number of likelihood profiling iterations for each
+#'   bound estimated
+#' @param ofvtol The relative tolerance for the objective function being close
+#'   enough to the `ofvIncrease`.
+#' @param paramDigits The number of significant digits required for the
+#'   parameter.  When interpolation attempts to get smaller than that number of
+#'   significant digits, it will stop.
+#' @param extrapolateExpand When extrapolating outside the range previously
+#'   tested, how far should the step occur as a ratio
+#' @returns A validated list of control options for log-likelihood profiling
+#' @family Profiling
+#' @seealso [profileLlp()]
+#' @export
+llpControl <- function(ofvIncrease = qchisq(0.95, df = 1),
```

Comment:
Because of `shinyMixR` this should now have a new method `rxUiDeparse.llpControl()`...  I will write one quickly.  I am trying to release this today and would like this included (I know I'm reviewing last minute...)